server/services/withings/api/withings.controller.js

```
@@ -0,0 +1,110 @@
+   * @apiName reset
+   * @apiGroup Withings
+   */
+  async function reset(req, res) {
+    await withingsHandler.deleteVar(req.user.id);
+    await withingsHandler.deleteDevices();
+    res.json({
+      success: true,
+    });
+  }
+
+  /**
+   * @description Build an authorization uri (to get authorizationcode).
+   * @api {post} /api/v1/service/withings/oauth2/client/authorization-uri Build an authorization uri
+   * @apiName BuildAuthorizationUri
+   * @apiGroup OAuth2
+   */
+  async function buildAuthorizationUri(req, res) {
+    if (req.body && req.body.service_id && req.body.integration_name && req.headers.referer) {
+      const authorizationUriResult = await withingsHandler.oauth2Client.buildAuthorizationUri(
+        req.body.service_id,
+        req.user.id,
+        req.body.integration_name,
+        req.headers.referer,
+      );
+      res.json({
+        authorizationUri: authorizationUriResult,
+      });
+    }
+  }
+
+  /**
+   * @description Get access token and save it in database (for current service).
+   * @api {post} /api/v1/service/withings/oauth2/client/access-token Build an getToken uri (to get token access)
+   * @apiName buildAccessTokenUri
+   * @apiGroup OAuth2
+   */
+  async function getAccessTokenUri(req, res) {
+    if (req.body && req.body.service_id && req.body.authorization_code && req.headers.referer) {
+      const authResult = await withingsHandler.oauth2Client.getAccessToken(
+        req.body.service_id,
+        req.user.id,
+        req.body.authorization_code,
+        req.headers.referer,
+      );
+      res.json({
+        result: authResult,
+      });
+    }
+  }
+
+  /**
+   * @description Return the current integration config.
+   * @api {get} /api/v1/service/withings/oauth2/client Return the current integration config.
+   * @apiName getCurrentConfig
+   * @apiGroup oauth2
+   */
+  async function getCurrentConfig(req, res) {
+    if (req.query && req.query.service_id) {
```

When a service_id is not sent, this will wait forever

---

I fix this

---

It's worse, it'll return null

If it's fine:
- It should return the response
- It should return an appropriate error (400 - Bad Request Or 422 - Unprocessable Entity)

---

I fix this with return status 400